### PR TITLE
Add energy demand to machine infotexts

### LINKED ISSUE #154: Use machines' infotext to display energy production/consumption
> Abdou31: What i would like to see is being able to know how much power a particular generator gives, like hovering an hv fuelfire generator doesn't tell you how much power it gives

### PR DESCRIPTION
**Pull request by Athozus**
Add a new line in infotext in `technic.register_base_machine()`, and the translations (based on other bigger translations that contains `Demand: @3`)

Partially implements #154

### CODE REVIEW

**Comment by S-S-X** [comment]
It seems like some possibly relevant stuff were removed from localization template here?

**Comment by Athozus** [comment]
> It seems like some possibly relevant stuff were removed from localization template here?

I don't know, I ran `i18n.py` (https://github.com/minetest-tools/update_translations/blob/master/i18n.py), the one I use in mail.

**Comment by S-S-X** [comment]
Seems more than just that, probably because it regenerates template file and translations used as source were incomplete.
Maybe you could try run it again without sort to get idea what it is actually going to drop, for one seems like "Chernobylite=" at least was also dropped.

Might be that some of those actually can be dropped but should check what it actually is.

Anyway either find out exact differences and check what should or shouldn't be removed or alternatively reset translation and add new stuff without using that tool.

**Comment by S-S-X** [comment]
... actually with sorting, seems like those were sorted but now are not anymore. Anyway, changes to translation files should be pretty much minimal. If other stuff with translations needs work maybe leave it out here and create new issue about it.

**Comment by S-S-X** [comment]
And maybe drop i18n.py, not sure if it was accidentally added now but we probably do not want to maintain it here anyway and for possible automatic run it could be just grabbed from repo where it is maintained.

**Comment by Athozus** [comment]
> And maybe drop i18n.py, not sure if it was accidentally added now but we probably do not want to maintain it here anyway and for possible automatic run it could be just grabbed from repo where it is maintained.

Accidentally added, yep

**Review by S-S-X — APPROVED** [review]
LGTM now, simple and good addition 👍

**Comment by OgelGames** [comment]
This only half fixes #154, it does not add the info for generators, and it also misses some machines (tool workshop, quarry, forcefield).

**Comment by Athozus** [comment]
> > This only half fixes #154, it does not add the info for generators, and it also misses machines that don't use the machine base (tool workshop, quarry, forcefield, maybe others).
> 
> There's simple fix for that: remove closing link. Then just create another PR for rest of it. Or of course can also update some of it here but I don't see any reason forcing that, going forward and allows trying it out even if it is just single machine.

What do I have to do so ?

**Comment by S-S-X** [comment]
> What do I have to do so ?

Basically, to close linked issue, just copy similar stuff to other machines and generators (that are not already using base machine registration).

**Comment by Athozus** [comment]
I did the edit that you wanted. Btw, while I was testing all machines to check if there is no error I noticed that it should be to show more parameter of machines (eg. the shape of the forcefield).

**Comment by S-S-X** [comment]
> Btw, while I was testing all machines to check if there is no error I noticed that it should be to show more parameter of machines (eg. the shape of the forcefield).

There's no really need to display everything on infotext, just some basic info.
Force field for example is usually configured just once for its purpose and then it stays like that so I guess probably in more than 99% of use cases it is not really important to be able to look it up as quickly as possible or multiple times.

**Comment by S-S-X** [comment]
Removed closing keyword and changed title a bit.

Change set size is very nice, it does single thing and I think it now covers all the machines in `technic` mod (also better to not include CNC in same commit as it is different mod, no need to worry about squashing).

Supply info, while similar, is another thing and could wait for another PR, possibly one that would also implement stuff mentioned in #177 comments.

Some minor style / formatting inconsistencies but I don't think that's a problem (but if you have extra time go ahead and add those missing tabs), I'd say good for merge either way.

**Review by OgelGames — CHANGES_REQUESTED** [review]
The lamp, LED and quarry use variables for their infotext, so the additions should be moved to those. While it doesn't impact performance that much, recreating the exact same infotext every time is a still a waste (especially for the quarry).

**Comment by Athozus** [comment]
> The lamp, LED and quarry use variables for their infotext, so the additions should be moved to those. While it doesn't impact performance that much, recreating the exact same infotext every time is a still a waste (especially for the quarry).
> 
> Otherwise everything looks good +1

I don't understand what changes you requested. Anyway, would be happy to do it ;)

**Comment by S-S-X** [comment]
> I don't understand what changes you requested. Anyway, would be happy to do it ;)

Exact same thing I suggested above for LV lamp: _(or in this case could probably just bake it all directly to active_desc variable)_

Basically, instead of reconstructing active description in technic_run callback do it in initialization (this is for LV lamp but similar elsewhere if possible):
```diff
+local demand = 50
 local active_desc = S("@1 Active", desc)  .. "\n" .. S("Demand: @1", technic.EU_string(demand))
 local unpowered_desc = S("@1 Unpowered", desc)
 local off_desc = S("@1 Off", desc)
-local demand = 50
```

It isn't significant performance wise but keeps function bit simpler increasing readability a bit.

**Comment by S-S-X** [comment]
Well, I guess it is already second approval based on last review (https://github.com/mt-mods/technic/pull/317#pullrequestreview-1542039879) and changes that addressed exactly/only what was requested.

File: technic/machines/HV/quarry.lua
Comment (OgelGames) [inline technic/machines/HV/quarry.lua:17]: Should be on the same line.

Comment (OgelGames) [reply technic/machines/HV/quarry.lua:17]: Wait, nvm, that would make it too long 🤦

**Comment by OgelGames** [comment]
Better to wait for approval for changes, but I fixed what I needed to anyway (https://github.com/mt-mods/technic/commit/13cdb0b410b21ba3873a02491b1fd0dd60b7918e)

**Comment by S-S-X** [comment]
> Better to wait for approval for changes, but I fixed what I needed to anyway ([13cdb0b](https://github.com/mt-mods/technic/commit/13cdb0b410b21ba3873a02491b1fd0dd60b7918e))

So minor style issues, based on discussion and change request I thought that's fine. Would have been nice to fix all of it at once then 🤷

**Comment by OgelGames** [comment]
Well really it was fine, but my OCD couldn't accept the ugly addition to the nice pretty code 😄

**Pull request merged**
